client/network/src/transport.rs

```diff
@@ -126,11 +126,18 @@ pub fn build_transport(
 
 			core::upgrade::apply(stream, upgrade, endpoint, upgrade::Version::V1)
 				.map_ok(|(id, muxer)| (id, core::muxing::StreamMuxerBox::new(muxer)))
-		})
+		});
 
-		.timeout(Duration::from_secs(20))
-		.map_err(|err| io::Error::new(io::ErrorKind::Other, err))
-		.boxed();
+	let transport = if cfg!(not(target_os = "unknown")) {
+		transport
+			.timeout(Duration::from_secs(20))
```

Comment:
Why not have the timeout with `unknown`? There's no other timeout anywhere in the stack, and no timeout means that we just continue trying to connect **forever** and never try a different address or a different node.

Reply:
Also panics because of `std::time::Instant::now()`.

Reply:
Is that still the case?

Reply:
It was the case when I tested it 4 hours ago, yeah.

Reply:
See https://github.com/libp2p/rust-libp2p/pull/1426.